handlers/producers.go

```diff
@@ -269,8 +285,23 @@ func (umh ProducersHandler) DestroyProducer(c *gin.Context) {
 		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "A producer with the given details was not found"})
 		return
 	}
-
-	logger.Info("Producer " + name + " has been deleted")
+	user := getUserDetailsFromMiddleware(c)
+	message := "Producer " + name + " has been deleted"
+	logger.Info(message)
+	var auditLogs []interface{}
+	newAuditLog := models.AuditLog{
+		ID:              primitive.NewObjectID(),
+		StationName:     stationName,
+		Message:       	 message,
+		CreatedByUser:   user.Username,
+		CreationDate:    time.Now(),
+		UserType: 		 user.UserType,
+	}
+	auditLogs = append(auditLogs, newAuditLog)
+	err = CreateAuditLogs(auditLogs)
+	if err != nil {
+		logger.Warn("CreateAuditLogs error: " + err.Error())
```

Comment:
The error happens in destroyProducer and not in CreateAuditLogs, change it across the entire places it appears